### ISSUE TITLE
respect TerminateOnNaN

### ISSUE THREAD

**Issue by jbarthelmes** [keras-team/keras-tuner#442]
# Expected
When I add `TerminateOnNaN` as a callback, training should stop after any batch when `isnan(loss)` via the callback setting `model.stop_training`.

# Actual
Training does not stop. Output is à la
```
Epoch 5/10
   1/1500 [..............................] - ETA: 0s - loss: 2.2759 - accuracy: 0.2188Batch 1: Invalid loss, terminating training
   2/1500 [..............................] - ETA: 46s - loss: nan - accuracy: 0.1719  Batch 2: Invalid loss, terminating training
   3/1500 [..............................] - ETA: 59s - loss: nan - accuracy: 0.1562Batch 3: Invalid loss, terminating training
   4/1500 [..............................] - ETA: 1:06 - loss: nan - accuracy: 0.1406Batch 4: Invalid loss, terminating training
   5/1500 [..............................] - ETA: 1:10 - loss: nan - accuracy: 0.1187Batch 5: Invalid loss, terminating training
   6/1500 [..............................] - ETA: 1:12 - loss: nan - accuracy: 0.1250Batch 6: Invalid loss, terminating training
   7/1500 [..............................] - ETA: 1:14 - loss: nan - accuracy: 0.1384Batch 7: Invalid loss, terminating training
```

# Affected Version
current git master (372989b), installed from pip .tar.gz, tested in google colab
tuner is Hyperband

# Reproducer
[on colab](https://colab.research.google.com/drive/1LWk-HJCnaYybMO-qYfL6VvrjT_j-vyRS?usp=sharing)

**Comment by parkournerd** [comment]
I get the same behaviour with tensorflow keras

**Comment by shun-lin** [comment]
I ran the colab linked above and I think that the model does indeed stopped when it get a nan value at the end of batch as expected right? (The above printing issue may be related to a different issue.

**Comment by jbarthelmes** [comment]
Seems fixed
